flanker-test-app/src/main/java/edu/northwestern/mobiletoolbox/flanker_app/flanker/step/flanker/FlankerFormStepView.java

```diff
@@ -0,0 +1,187 @@
+import androidx.annotation.Nullable;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import org.sagebionetworks.research.domain.form.interfaces.InputField;
+import org.sagebionetworks.research.domain.step.interfaces.Step;
+import org.sagebionetworks.research.presentation.DisplayString;
+import org.sagebionetworks.research.presentation.mapper.DrawableMapper;
+import org.sagebionetworks.research.presentation.model.ColorThemeView;
+import org.sagebionetworks.research.presentation.model.ImageThemeView;
+import org.sagebionetworks.research.presentation.model.action.ActionView;
+import org.sagebionetworks.research.presentation.model.implementations.UIStepViewBase;
+import org.sagebionetworks.research.presentation.model.interfaces.UIStepView;
+
+public class FlankerFormStepView implements UIStepView {
+
+    @NonNull
+    private final FlankerFormStep flankerFormStep;
+
+    @NonNull
+    private final UIStepView uiStepView;
+
+    @NonNull
+    public static FlankerFormStepView fromFlankerFormStep(@NonNull Step step, @NonNull DrawableMapper mapper) {
+        UIStepView uiStepView = UIStepViewBase.fromUIStep(step, mapper);
+
+        FlankerFormStep flankerFormStep = (FlankerFormStep) step;
+
+        return new FlankerFormStepView(flankerFormStep, uiStepView);
+    }
+
+    public FlankerFormStepView(@NonNull final FlankerFormStep flankerFormStep,
+            @NonNull final UIStepView uiStepView) {
+        this.flankerFormStep = flankerFormStep;
+        this.uiStepView = uiStepView;
+    }
+
+    @Override
+    @Nullable
+    public ActionView getActionFor(final String actionType) {
+        return uiStepView.getActionFor(actionType);
+    }
+
+    @Override
+    @NonNull
+    public ImmutableMap<String, ActionView> getActions() {
+        return uiStepView.getActions();
+    }
+
+    @Override
+    @Nullable
+    public ColorThemeView getColorTheme() {
+        return uiStepView.getColorTheme();
+    }
+
+    @Override
+    @Nullable
+    public DisplayString getDetail() {
+        return uiStepView.getDetail();
+    }
+
+    @Override
+    @Nullable
+    public DisplayString getFootnote() {
+        return uiStepView.getFootnote();
+    }
+
+    @Override
+    @Nullable
+    public ImageThemeView getImageTheme() {
+        return uiStepView.getImageTheme();
+    }
+
+    @Override
+    @Nullable
+    public DisplayString getText() {
+        return uiStepView.getText();
+    }
+
+    @Override
+    @Nullable
+    public DisplayString getTitle() {
+        return uiStepView.getTitle();
+    }
+
+    @Override
+    @NonNull
+    public String getIdentifier() {
+        return uiStepView.getIdentifier();
+    }
+
+    @NonNull
+    @Override
+    public String getType() {
+        return FlankerFormStep.TYPE;
+    }
+
+    @NonNull
+    public ImmutableList<InputField> getInputFields() {

```

Comment:
Note that this returns an `ImmutableList<InputField>`. FormUIStepView returns an `ImmutableList<InputFieldView>`, I tried going down that route and it was a lot of code to convert everything used by `InputField`, `Choice`, etc

This is why this class extends `UIStepView`, instead of `FormUIStepView`.